source/extensions/filters/http/ext_authz/config.cc

```diff
@@ -19,22 +20,39 @@ namespace ExtAuthz {
 Http::FilterFactoryCb ExtAuthzFilterConfig::createFilterFactoryFromProtoTyped(
     const envoy::config::filter::http::ext_authz::v2alpha::ExtAuthz& proto_config,
     const std::string&, Server::Configuration::FactoryContext& context) {
-  auto filter_config =
+
+  const auto filter_config =
       std::make_shared<FilterConfig>(proto_config, context.localInfo(), context.scope(),
                                      context.runtime(), context.clusterManager());
-  const uint32_t timeout_ms = PROTOBUF_GET_MS_OR_DEFAULT(proto_config.grpc_service(), timeout, 200);
+
+  if (proto_config.has_http_service()) {
+    const uint32_t timeout_ms = PROTOBUF_GET_MS_OR_DEFAULT(proto_config.http_service().server_uri(),
+                                                           timeout, DefaultTimeout);
+    return [
+      filter_config, timeout_ms, cluster_name = proto_config.http_service().server_uri().cluster(),
+      path_prefix = proto_config.http_service().path_prefix()
+    ](Http::FilterChainFactoryCallbacks & callbacks) {
+      auto client = std::make_unique<Filters::Common::ExtAuthz::RawHttpClientImpl>(
+          cluster_name, filter_config->cm(), std::chrono::milliseconds(timeout_ms), path_prefix);
+      callbacks.addStreamDecoderFilter(Http::StreamDecoderFilterSharedPtr{
```

Comment:
Do you need to construct a shared pointer around the result of `make_shared` here?

Reply:
👀